mapstruct-field-mapping/src/test/java/org/mapstruct/example/CustomerMapperTest.java

```diff
@@ -0,0 +1,80 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.example;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+import org.junit.Test;
+import org.mapstruct.example.dto.Customer;
+import org.mapstruct.example.dto.CustomerDto;
+import org.mapstruct.example.dto.OrderItem;
+import org.mapstruct.example.dto.OrderItemDto;
+import org.mapstruct.example.mapper.CustomerMapper;
+
+/**
+ *
+ * @author Filip Hrisafov
+ */
+public class CustomerMapperTest {
+
+    @Test
+    public void testMapDtoToEntity() {
+
+        CustomerDto customerDto = new CustomerDto();
+        customerDto.id = 10L;
+        customerDto.customerName = "Filip";
+        OrderItemDto order1 = new OrderItemDto();
+        order1.name = "Table";
+        order1.quantity = 2L;
+        customerDto.orders = new ArrayList<>( Collections.singleton( order1 ) );
+
+        Customer customer = CustomerMapper.MAPPER.toCustomer( customerDto );
+
+        assertThat( customer.getId() ).isEqualTo( 10 );
+        assertThat( customer.getName() ).isEqualTo( "Filip" );
+        assertThat( customer.getOrderItems() )
+            .extracting( OrderItem::getName, OrderItem::getQuantity )
+            .containsExactly( tuple( "Table", 2L ) );
+    }
+
+    @Test
+    public void testEntityDtoToDto() {
+
+        Customer customer = new Customer();
+        customer.setId( 10L );
+        customer.setName( "Filip" );
+        OrderItem order1 = new OrderItem();
+        order1.setName( "Table" );
+        order1.setQuantity( 2L );
+        customer.setOrderItems( Collections.singleton( order1 ) );
+
+        CustomerDto customerDto = CustomerMapper.MAPPER.fromCustomer( customer );
+
+        assertThat( customerDto.id ).isEqualTo( 10 );
+        assertThat( customerDto.customerName ).isEqualTo( "Filip" );
+        assertThat( customerDto.orders )
```

Comment:
You probably meant about the extracting assertions and not this line. In any case I changed it by using the name of the properties in the extraction. Is that fine with you?